sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -730,6 +730,8 @@ class Analyzer(
   case class ResolveNamespace(catalogManager: CatalogManager)
     extends Rule[LogicalPlan] with LookupCatalog {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case UnresolvedNamespace(Seq()) =>
+        ResolvedNamespace(currentCatalog.asNamespaceCatalog, Seq.empty[String])
```

Comment:
@cloud-fan this was introduced to handle commands where namespace is used as optional  (`SHOW NAMESPACES` for example). Is this approach (which treats empty `Seq` as `None`) reasonable?

Reply:
This may not be a good idea since the user may need an extra check to deduce whether empty `Seq` is a `None`. For example, I have to do additional check for the following command:
```scala
    case ShowTablesStatement(NonSessionCatalogAndNamespace(catalog, ns), pattern) =>
      val namespace = if (ns.isEmpty && currentCatalog.name.equals(catalog.name)) {
        catalogManager.currentNamespace.toSeq
      } else { ns }
      ShowTables(catalog.asTableCatalog, namespace, pattern)
```

Reply:
`CatalogAndNamespace` doesn't look up the namespace, but look up the catalog. I think it can handle Nil, which resolves catalog to the current catalog, and return Nil as the namespace identifier.

Reply:
The conflict here is that `SHOW NAMESPACES` treats `None` as `Nil`, but `SHOW TABLES` treats `None` as `current namespace`, thus causing the ambiguity. To make `SHOW TABLES` work with `Nil` approach (the current one), I have to do the following:
```scala
    case ShowTablesStatement(NonSessionCatalogAndNamespace(catalog, ns), pattern) =>
      val namespace = if (ns.isEmpty && currentCatalog.name.equals(catalog.name)) {
        catalogManager.currentNamespace.toSeq
      } else { ns }
      ShowTables(catalog.asTableCatalog, namespace, pattern)
```

Reply:
I think `SHOW TABLES` is a special case, can we manually handle it in `ResolveNamespace`?

Reply:
OK, will do.